www/js/app.js

```diff
@@ -1422,35 +1530,109 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
                     anchor.target = '_blank';
                 } else {
                     // It's a link to an article or file in the ZIM
-                    var uriComponent = uiUtil.removeUrlParameters(href);
-                    var contentType;
-                    var downloadAttrValue;
-                    // Some file types need to be downloaded rather than displayed (e.g. *.epub)
-                    // The HTML download attribute can be Boolean or a string representing the specified filename for saving the file
-                    // For Boolean values, getAttribute can return any of the following: download="" download="download" download="true"
-                    // So we need to test hasAttribute first: see https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute
-                    // However, we cannot rely on the download attribute having been set, so we also need to test for known download file types
-                    var isDownloadableLink = anchor.hasAttribute('download') || regexpDownloadLinks.test(href);
-                    if (isDownloadableLink) {
-                        downloadAttrValue = anchor.getAttribute('download');
-                        // Normalize the value to a true Boolean or a filename string or true if there is no download attribute
-                        downloadAttrValue = /^(download|true|\s*)$/i.test(downloadAttrValue) || downloadAttrValue || true;
-                        contentType = anchor.getAttribute('type');
-                    }
-                    // Add an onclick event to extract this article or file from the ZIM
-                    // instead of following the link
-                    anchor.addEventListener('click', function (e) {
-                        var zimUrl = uiUtil.deriveZimUrlFromRelativeUrl(uriComponent, baseUrl);
-                        goToArticle(zimUrl, downloadAttrValue, contentType);
-                        e.preventDefault();
-                    });
+                    addListenersToLink(anchor, href);
                 }
             });
+            // Add event listeners to the main document so user can open current document in new tab or window
+            addListenersToLink(articleDocument, encodeURIComponent(dirEntry.url.replace(/[^/]+\//g, '')));
         }
-        
+
```

Comment:
Move funciton below to `uiUtil.js` to de-clutter `app.js`.

Reply:
Actually, having tried this (a bit) it turns out to be quite tricky, because the function needs access to a fair amount of state that is stored in app.js, and needs to set crucial variables such as `articleContainer` and `articleWindow`. It also needs to access the function `goToArticle()` in `app.js`, and for architectural reasons RequireJS won't allow `app` to be a dependency of `uiUtil`. While it would be possible to pass variables and functions needed, and we could call back to app.js with the values to set in global variables, it becomes a bit unwieldy, and maybe exceeds the definition of UI 'utility'.